### PR TITLE
request pagination

### PR DESCRIPTION
**Pull request by gbrvalerio**
- This draft PR creates the basis for implementing Pagination in storage requests

#### Core
- [x] Create Pagination class
- [x] Adapt Request class to include pagination
- [x] Adapt DatabaseStorage to handle the request pagination
- [x] Remove the limit prop from the Order class
- [x] Fix current usage of the Request class to use Pagination

#### Implementation
- [ ] Add skip and limit url params to the list devices endpoint
  - [ ] Update Docs
  - [x] Update query
- [ ] Add skip and limit url params to the list users endpoint
  - [ ] Update Docs
  - [x] Update query
- [ ] Add skip and limit url params to the base class SimpleObjectResource (affects Groups, Calendars, Orders)
  - [ ] Update Docs
  - [x] Update query
- [ ] Add skip and limit url params to the list drivers endpoint
  - [ ] Update Docs
  - [x] Update query
  
### Databases Tested
- [x] Postgres
- [x] H2
- [ ] MS SQL

@tananaev

### CODE REVIEW

File: src/main/java/org/traccar/storage/DatabaseStorage.java [hunk 1]
```diff
@@ -307,14 +309,25 @@ private String formatOrder(Order order) {
...
+            	if (databaseType.equals("Microsoft SQL Server")) {
                     result.append(" OFFSET 0 ROWS FETCH FIRST ");
-                    result.append(order.getLimit());
+                    result.append(pagination.getLimit());
                     result.append(" ROWS ONLY");
                 } else {
                     result.append(" LIMIT ");
-                    result.append(order.getLimit());
+                    result.append(pagination.getLimit());
                 }
```
Comment (tananaev) [inline src/main/java/org/traccar/storage/DatabaseStorage.java:331]: Which databases have you tested?

Comment (gbrvalerio) [reply src/main/java/org/traccar/storage/DatabaseStorage.java:331]: PG so far

Comment (tananaev) [reply src/main/java/org/traccar/storage/DatabaseStorage.java:331]: We need to test at least H2, MySQL, Postgres and MS SQL. Is there an SQL standard for it?

Comment (gbrvalerio) [reply src/main/java/org/traccar/storage/DatabaseStorage.java:331]: code wise, the path is correct?

Comment (tananaev) [reply src/main/java/org/traccar/storage/DatabaseStorage.java:331]: I only did a brief look, but I think it makes sense. I would separate database changes and report changes though.

**Comment by gbrvalerio** [comment]
@tananaev regarding docs: is it automatically updated or should I do some manual work here? If so, can you point me where should I pay attention and update?

ty

**Comment by tananaev** [comment]
What docs do you need to update?

**Comment by gbrvalerio** [comment]
the api ones to include the new `skip` and `limit` url params in the endpoints

**Comment by tananaev** [comment]
Well, we need to split the PR first.

**Comment by gbrvalerio** [comment]
closed in favor of #5208 and #5209

**Pull request closed without merging**